### PR TITLE
CDAP-14203 Add more logging when RemoteProcessController encounters uncommon failure scenarioes

### PR DESCRIPTION
**Pull request by anwar6953**
https://issues.cask.co/browse/CDAP-14203
Add more logging when RemoteProcessController encounters uncommon failure scenarios.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/remote/SSHRemoteProcessController.java [hunk 1]
```diff
@@ -69,6 +69,9 @@ public boolean isRunning() throws Exception {
       ByteStreams.toByteArray(process.getErrorStream());
 
       int exitCode = process.waitFor();
+      if (exitCode != 0) {
+        LOG.info("Received exit code {} when checking for remote process.", exitCode);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/remote/SSHRemoteProcessController.java]: Logs with the `programRunId`

Comment (anwar6953) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/remote/SSHRemoteProcessController.java]: done

**Comment by anwar6953** [comment]
https://builds.cask.co/browse/CDAP-DUT6616

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/remote/SSHRemoteProcessController.java (hunk 1 above)
Comment (anwar6953) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/remote/SSHRemoteProcessController.java]: @chtyim - can you take another look?

**Review by chtyim — APPROVED** [review]
LGTM

**Pull request merged**